### DISCUSSION TITLE
Multiple auth.users rows with the same email address

### DISCUSSION THREAD

**Discussion by techpet** [nhost/nhost#1071] [Q&A]
I am working in a scenario where a user can create accounts in many different organizations. In this scenario, the user may register with the same email (but possibly different passwords-basically creating new accounts) in various organizations. The problem is that in `auth.users` table, the email field is set as a unique field and so there could not exist multiple rows with the same email. 

In theory, one could modify the `auth.users` table by removing the `unique` attribute from `email` field but is it a good practice? Would nhost sdk work correctly after that? 

Finally for implementing the registration of accounts for the various different organizations, one could also store the organization id in the `auth.users.metadata` field using `signUp()`. However, during login, how the organization id be passed in the `signIn()` sdk function in order to retrieve the correct account (row in `auth.users` table)?

What would be the recommended way to achieve such a functionality overall?

**Comment by elitan** [comment]
> In theory, one could modify the auth.users table by removing the unique attribute from email field but is it a good practice? Would nhost sdk work correctly after that?

You should not change the `auth` (or `storage`) schemas because it can break authentication completely. 

The way we've solved this with Nhost is the following:

```
auth.users
- id
- display_name

public.workspaces (organization)
- id
- name

public.workspace_members
- id
- workspace_id
- user_id
- type (owner or member)
- (Here, you can add extra metadata that is specific for the user for this particular workspace)
```

Maybe this gives you some ideas for solving your particular use case?

**Reply by techpet** [reply]
Thank you @elitan ! Ι've seen this approach in another topic also and this is similar to what I've implemented now. 

The problem is that either one won't be able to use the nhost sdk afterwards and have to reimplement all the hasura-auth functionality for the workspace_members table (instead of auth.users) or one goes after the nhost sdk and loses the ability to signup users with an already present email address in the auth.users table.

Is the above true or is there an easy customization of the signUp and signIn nhost functions?

How did you solve the problem of a user registering to different workspaces with the same email but different password in your case?

**Reply by elitan** [reply]
In our case, we don't have the concept of different users. There is only one user. That one user can belong to different workspaces.

If we want to attach unique properties to a user and a particular workspace, we use the `workspace_members` table. For example, a user can be an owner of one workspace and a member of another workspace.

**Reply by techpet** [reply]
After some thought, going with the passwordless login and using one account per user along with your proposed approach will eventually work for my case also! So each time you introduce a new role for a user (even on a different workspace) do you also update the `user_roles` table (to update `x-hasura-allowed-roles` claim) and afterwards use appropriate `x-hasura-role` in order to authorize the user?
